pkg/services/ngalert/state/manager.go

```diff
@@ -261,6 +253,59 @@ func (st *Manager) setNextState(ctx context.Context, alertRule *ngModels.AlertRu
 	return nextState
 }
 
+// PauseStates sets all states from an alert rule to pause. If states need to be resolved it tries to take an image.
+func (st *Manager) PauseStates(ctx context.Context, r *ngModels.AlertRule) []StateTransition {
```

Comment:
I think this has a lot of duplication with `deleteStaleStatesFromCache`, perhaps we can use the same code? Both take an `*AlertRule` so we know if the rule is paused or not?